testing/test_request.go

```diff
@@ -0,0 +1,81 @@
+package testing
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	contractstesting "github.com/goravel/framework/contracts/testing"
+	"github.com/goravel/framework/support/collect"
+	"github.com/goravel/framework/support/maps"
+)
+
+type TestRequest struct {
+	t              *testing.T
+	defaultHeaders map[string]string
+	defaultCookies map[string]string
+}
+
+func NewTestRequest(t *testing.T) contractstesting.TestRequest {
+	return &TestRequest{
+		t:              t,
+		defaultHeaders: make(map[string]string),
+		defaultCookies: make(map[string]string),
+	}
+}
+
+func (m *TestRequest) WithHeaders(headers map[string]string) contractstesting.TestRequest {
```

Comment:
We can use `r` in all structs (consistent with other structs), except test suit (it's `s`).

```suggestion
func (r *TestRequest) WithHeaders(headers map[string]string) contractstesting.TestRequest {
```